frontend/lib/framework/ID.cpp

```diff
@@ -142,7 +156,8 @@ ID ID::parentSymbolId(Context* context) const {
   }
 
   if (this->isFabricatedId() &&
-      this->fabricatedIdKind() == FabricatedIdKind::Generated) {
+      this->fabricatedIdKind() == FabricatedIdKind::Generated &&
+      !isSymbolDefiningScope()) {
```

Comment:
If I understand correctly, this assumes that it's not possible to have a generated ID within another generated ID. That's probably fine, but needs to be documented as a constraint somewhere. Potentially as a `CHPL_ASSERT` somewhere.

Reply:
Technically the assumption is that generated uAST will not contain nested scope-defining symbols. I've pushed a commit containing both comments and an assertion to enforce this.